Neos.Neos/Classes/FrontendRouting/EventSourcedFrontendNodeRoutePartHandler.php

```diff
@@ -223,7 +228,20 @@ public function matchWithParameters(&$requestPath, RouteParameters $parameters)
                 return false;
             }
         } catch (NodeNotFoundException $exception) {
-            // we silently swallow the Node Not Found case, as you'll see this in the server log if it interests you
+            // if there are fundamental errors with your dimension config (mismatch between database and settings), people only see a 404 - which is a problem because it
+            // is not possible for people to analyze this further. This is why in the error case, we check whether the root node exists in the expected DimensionSpacePoint, because
+            // this way, we can give some hints on what to do.
+            $this->securityContext->withoutAuthorizationChecks(function () use ($exception, $dimensionSpacePoint, $contentRepository) {
+                $sitesNodeAggregate = $contentRepository->getContentGraph(WorkspaceName::forLive())->findRootNodeAggregateByType(NodeTypeNameFactory::forSites());
+                if ($sitesNodeAggregate === null) {
+                    throw new NodeNotFoundException('No Sites Node found in the Content Graph. Please run the setup tool at /setup, which describes the steps to import a site.', 1762418159, $exception);
+                }
+                if (!$sitesNodeAggregate->coversDimensionSpacePoint($dimensionSpacePoint)) {
+                    throw new NodeNotFoundException('The Neos.Neos:Sites root node does not cover the resolved dimension space point ' . $dimensionSpacePoint->toJson() . ". This can happen for the following reasons: \n - 1. dimension config in Neos.ContentRepositoryRegistry.contentRepositories.default.contentDimensions does not match Site Dimension Routing Configuration in Neos.Neos.sites.*.contentDimensions.resolver \n - 2. configured dimension configuration does not match the dimensions in the database.\n\n To debug this further, see https://docs.neos.io/guide/content-repository/content-dimensions/migrating-dimension-config or run the setup tool at /setup, which contains further diagnostics.", 1762418208, $exception);
```

Comment:
```suggestion
                    throw new NodeNotFoundException('The Neos.Neos:Sites root node does not cover the resolved dimension space point ' . $dimensionSpacePoint->toJson() . ". This can happen for the following reasons: \n - 1. dimension config in Neos.ContentRepositoryRegistry does not match site dimension routing configuration in Neos.Neos.sites \n - 2. configured dimension configuration does not match the dimensions in the database.\n\n To debug this further, see https://docs.neos.io/guide/content-repository/content-dimensions/migrating-dimension-config or run the setup tool at /setup, which contains further diagnostics.", 1762418208, $exception);
```

I tested this in a multi cr setup so the settings were one of but i think the hint is enought and `/setup` gives good information about the actual path or we also do the lookup here to find the actual location